#!/usr/bin/env python
PACKAGE = "hope"
from dynamic_reconfigure.parameter_generator_catkin import *

# name - a string which specifies the name under which this parameter should be stored
# type - defines the type of value stored, and can be any of int_t, double_t, str_t, or bool_t
# level - A bitmask which will later be passed to the dynamic reconfigure callback.
#         When the callback is called all of the level values for parameters that have been changed are ORed together and the resulting value is passed to the callback.
# description - string which describes the parameter
# default - specifies the default value
# min - specifies the min value (optional and does not apply to strings and bools)
# max - specifies the max value (optional and does not apply to strings and bools)
gen = ParameterGenerator()
gen.add("min_height", double_t, 0, "min_height", -10., -100., 100.)
gen.add("max_height", double_t, 0, "max_height", 10., -100., 100.)
gen.add("min_depth", double_t, 0, "min_depth", 0., 0., 10.)
gen.add("max_depth", double_t, 0, "max_depth", 1.5, 0.3, 100.)

exit(gen.generate(PACKAGE, "hope", "hope"))